app/src/main/java/com/github/wnder/guessLocation/GuessLocationActivity.java

```diff
@@ -411,27 +440,134 @@ private void confirmButton() {
         ((FloatingActionButton) findViewById(R.id.confirmButton)).setImageResource(R.drawable.ic_baseline_list_24);
 
         //Send guess and update karma
-        if (!pictureID.equals(Utils.UNINITIALIZED_ID) && !(user instanceof GuestUser)) {
+        if (!picToGuess.getUniqueId().equals(Utils.UNINITIALIZED_ID) && !(user instanceof GuestUser)) {
             Location guessedLocation = new Location("");
             guessedLocation.setLatitude(guessPosition.getLatitude());
             guessedLocation.setLongitude(guessPosition.getLongitude());
             MapBoxHelper.onMapSnapshotAvailable(this.getApplicationContext(), guessPosition, picturePosition, (mapSnapshot) -> {
-                picturesDb.sendUserGuess(pictureID, user.getName(), guessedLocation, mapSnapshot);
+                picturesDb.sendUserGuess(picToGuess.getUniqueId(), user.getName(), guessedLocation, mapSnapshot);
             });
         }
-        picturesDb.updateKarma(pictureID, 1);
+        picturesDb.updateKarma(picToGuess.getUniqueId(), 1);
 
-        showActualLocation();
+        showActualLocation(computeScoreText(), true);
+
+        //Animate the next guess button
+        nextGuessButton.setVisibility(VISIBLE);
+        Animation button_animation = AnimationUtils.loadAnimation(this, R.anim.next_guess_button_anim);
+        nextGuessButton.startAnimation(button_animation);
+    }
+
+    /**
+     * Goes to the next guest preview activity
+     */
+    private void nextGuess() {
+        Intent intent = new Intent(this, GuessPreviewActivity.class);
+        startActivity(intent);
+        finish();
+    }
+
+    private void confirmTourMode() {
+        TextView scoreText = findViewById(R.id.scoreText);
+
+        if (!guessPossible) {
+            //Show pop up warning about skipping the picture
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setCancelable(true);
+            builder.setTitle(R.string.tour_mode_confirm_while_far_title);
+            builder.setMessage(R.string.tour_mode_confirm_while_far_text);
+
+            //What to do when OK is pressed
+            builder.setPositiveButton("Yes", (DialogInterface dialog, int which) -> {
+                if (tourIndex >= tourIDs.size()) {
+                    guessPossibleConfirmTourMode();
+                } else {
+                    tourIndex += 1;
+                    nextGuessTourMode();
+                }
+            });
```

Comment:
Design wise, not having a cancel button is really confusing

Reply:
I thought the return button would be the one of the phone by default. isn't that enough?

Reply:
It is okay for now but not intuitive for the user